src/org/jetbrains/java/decompiler/main/decompiler/ConsoleDecompiler.java

```diff
@@ -247,6 +247,9 @@ public void createArchive(String path, String archiveName, Manifest manifest) {
 
   @Override
   public void saveDirEntry(String path, String archiveName, String entryName) {
+    if (entryName.lastIndexOf('/') != entryName.length() - 1) {
+      throw new IllegalArgumentException(entryName + " is not a valid folder name. Folders have to end with a slash otherwise archive managers will treat them as files.");
+    }
     saveClassEntry(path, archiveName, null, entryName, null);
```

Comment:
Wouldn't it be better to just add the `'/'` in the entry name passed to `saveClassEntry`? I'd argue it's `saveDirEntry`s job to make sure it's actually saving it as a directory.